_posts/2022-12-21-theres-an-impostor-among-us/theres-an-impostor-among-us.Rmd

```diff
@@ -0,0 +1,180 @@
+
+Is there a way to _automatically_ detect those outliers?
+
+
+## Let's try using distances!
+
+`TODO`
+
+## What if we wanted to try out unsupervised approaches?
+
+A common unsupervised anomaly detection technique is the [k-means algorithm](https://www.naftaliharris.com/blog/visualizing-k-means-clustering/). The intuition behind k-means is that _each point in a cluster should be near the center of that cluster_.
+
+In R, we can compute the k-means clusters using the `kmeans` function and then visualize the results using the `factoextra` package. Below, we cluster the data into _3 clusters_^[Why 3 clusters? Why not 4 or 5 or 2?].
+
+```{r echo = TRUE}
+library(factoextra)
+kmean <- kmeans(scale(data), 3, nstart = 25) # Uses Euclidean distance
+fviz_cluster(kmean,
+             data = scale(data),
+             palette = c("#2E9FDF", "#00AFBB", "#E7B800"),
+             geom = "point",
+             ellipse.type = "convex",
+             ggtheme = theme_bw())
+```
+
+Notice the blue circle that appears in the upper left corner! What does it mean? Which data point does it represent? The `fviz_cluster` function will plot the data using the first and second principal components. If we wanted to, we could plot our data using 2 variables:
+
+```{r echo = TRUE}
+fviz_cluster(kmean,
+             data = data %>% select(vital_sbpdiastolic, vital_sbpsystolic),
+             palette = c("#2E9FDF", "#00AFBB", "#E7B800"),
+             geom = "point",
+             ellipse.type = "convex",
+             ggtheme = theme_bw())
+```
+
+It's a bit hard to tell, but in the plot where we only show the diastolic and systolic blood pressures, the blue circle appears in the cluster of yellow squares...
+
+Cool! Now, what if I want to try a different unsupervised approach? Like... DBSCAN?^[The best way to describe how DBSCAN works is with a visual: https://www.naftaliharris.com/blog/visualizing-dbscan-clustering/] 
+
+How well does DBSCAN work on our vitals data? We can use the `fpc` library to run the DBSCAN algorithm.
+
+```{r}
+library(fpc) # Uses Euclidean distance
+# You can calculate the distance independently and give that as input to the function
+db_results <- fpc::dbscan(data, eps = 1, MinPts = 4, scale = TRUE)
+fviz_cluster(db_results, 
+             scale(data),
+             geom = "point",
+             ggtheme = theme_bw())
+```
+
+The black dots correspond to the data points that could not be mapped to a cluster, according to the DBSCAN algorithm. These would be considered outliers.
+
+Now let's see what the results look like if we focus on just the blood pressure variables:
+
+```{r preview=TRUE}
+fviz_cluster(db_results, 
+             data %>% select(vital_sbpdiastolic, vital_sbpsystolic),
+             geom = "point",
+             ggtheme = theme_bw())
+```
+
+
+## What if we tried a supervised approach?
+
+`TODO`
+
+## Additional resources
```

Comment:
@derekbeaton @neilmistry are the resources below accurate? I copied what was listed in https://docs.google.com/document/d/1xXc_coelpyLRVxF4EvuGU2elA7B71VGuGVmIWovbqdY/edit